test/cockroachdb_test.go

```diff
@@ -0,0 +1,82 @@
+package helm_testing
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/stretchr/testify/require"
+	rbacv1 "k8s.io/api/rbac/v1"
+)
+
+func Test_Role(t *testing.T) {
+	helmChartPath, err := filepath.Abs("../cockroachdb")
+	require.NoError(t, err)
+	require.NotNil(t,helmChartPath)
+
+	t.Run("Test TLS Enabled but certs not provided", func(t *testing.T) {
+		options := &helm.Options{
+			SetValues: map[string]string{
+				"tls.enabled": "True",
+				"tls.certs.provided": "False",
+				"tls.certs.certManager": "False",
+			},
+			KubectlOptions: k8s.NewKubectlOptions("", "", "test-deployment"),
+		}
+
+		releaseName := "cockroachdb-test"
+
+		output := helm.RenderTemplate(t, options, helmChartPath, releaseName, []string{"templates/role.yaml"})
+		require.NotNil(t, output)
+
+		var role rbacv1.Role
+		helm.UnmarshalK8SYaml(t, output, &role)
+
+		verbs := role.Rules[0].Verbs
+
+		require.Len(t, verbs, 2)
+		require.Contains(t, verbs, "get")
```

Comment:
If you add a test like the below, it could help with making it more clear what resource the verbs are actually referring to in this test.

```
resources := role.Rules[0].Resources
require.Len(t, resources, 1)
require.Contains(t, resources, "secrets")
```